torch/csrc/autograd/python_variable.cpp

```diff
@@ -2140,10 +2140,12 @@ static int THPVariable_subclass_traverse(
           }
         }
       }
-
-      for (const auto& hook : torch::autograd::impl::hooks(tensor)) {
-        if (auto pyhook = dynamic_cast<PyFunctionTensorPreHook*>(hook.get())) {
-          Py_VISIT(pyhook->dict);
+      if (autograd_meta) {
```

Comment:
Why is this needed? The torch::autograd::impl::hooks call used to handle the case without autograd_meta but doesn't anymore?

Reply:
There's no reason to call hooks if there is no autograd_meta. I felt like it would be clearer if we enforce that when we call it. I don't feel strongly about this though.